mypy/checker.py

```diff
@@ -2868,9 +2882,14 @@ def infer_variable_type(self, name: Var, lvalue: Lvalue,
 
             self.set_inferred_type(name, lvalue, init_type)
 
-    def infer_partial_type(self, name: Var, lvalue: Lvalue, init_type: Type) -> bool:
+    def infer_partial_type(self, name: Var, lvalue: Lvalue, init_type: Type,
+                           init_expr: Expression) -> bool:
         init_type = get_proper_type(init_type)
         if isinstance(init_type, NoneType):
+            if isinstance(init_expr, CallExpr):
```

Comment:
In PR description, you say it is only OK for generic functions, is this still true? Or a different error will be shown for non-generic? Add a more detailed comment here maybe.

Reply:
For functions that always return `None` we generate error about a called function not returning a value, which happens elsewhere and isn't related to this code, which only deals with type inference.